### ISSUE TITLE
IE11 Syntax Error

### ISSUE THREAD

**Issue by warlord0** [xkjyeah/vue-google-maps#456]
I'm trying to use maps in a Laravel app using Vue.js. It works just great in Chrome including it in my app.js

```javascript
/**
 * VueGoogleMaps can't get it working in IE11 :(
 */
import * as VueGoogleMaps from 'vue2-google-maps'
Vue.use(VueGoogleMaps, {
  load: {
    key: 'mykey',
    libraries: 'places'
  }
})
```
In my component I add it and can add markers, move the map etc. all great.
```vue
        <GmapMap
          id="goomap"
          ref="map"
          :center="{lng: -1.2049487, lat: 52.7590669}"
          :zoom="18"
          class="col-md-8"
          style="min-height: 500px;"
        >
          <GmapMarker
            v-for="(m, index) in markers"
            :key="index"
            :position="m.position"
            :clickable="true"
            :draggable="true"
            @click="center=m.position"
          />
        </GmapMap>
```
But when I look in IE11 I get `SCRIPT1002: Syntax Error` in my compiled `app.js`.

After a lot of smashing head on desk I decided to separate out the `.js` by copying the asset in and pointing directly to the `.js` in my page. Then commenting out the `import` in my `app.js`.
```html
<script src="/js/vue-google-maps.js"></script>
```
Again, works fine in chrome. But in IE11 at least now I see the error is in `vue-google-maps.js` unfortunately the minified version :(

How do I build a non-minified version so I can pick up the error more specifically?

**Comment by warlord0** [comment]
Duplicate #452

**Comment by michtio** [comment]
I also get a syntax error when building a production build through webpack , polyfills are used, everything works, just google maps isn't working.

When I check the minified code I can find that there is 1 `let` statement which also uses arrow functions. Which it shouldn't of course.

**Comment by michtio** [comment]
I did solve it by adding the targets-webpack-plugin

```
.webpackConfig(webpack => {
        return {
            resolve: { extensions: ['.js'] },
            plugins: [
                new TargetsPlugin({
                    browsers: ['last 2 versions', 'chrome >= 41', 'IE 11'],
                }),
            ],
        };
    })
```

This builds perfectly fine and works in IE11

**Comment by warlord0** [comment]
Ooh gonna have to give that a go.

I got a bit further and the problem seems to be in lazy-value:

```javascript
module.exports = fn => { // This is where IE11 stops
	let called = false;
	let ret;

	return () => {
		if (!called) {
			called = true;
			ret = fn();
		}

		return ret;
	};
};
```

**Comment by warlord0** [comment]
@michtio you sir get upgraded to a love from a thumb

I mucked about in my `webpack.mix.js` file (Laravel's layer of webpack) and added in the targets plugin as the snippet below. No more problem with IE11.

```
$ yarn add targets-webpack-plugin -D
```

```javascript
const TargetsPlugin = require('targets-webpack-plugin')
...
mix.webpackConfig({
  plugins: [
    new TargetsPlugin({
      browsers: ['last 2 versions', 'chrome >= 41', 'IE 11'],
    }),
  ]
})
```

**Comment by michtio** [comment]
I spent a great deal figuring it out myself! @warlord0 glad to share :)

**Comment by hambroz** [comment]
I hate to say this but I really do not see this as a solution but rather a patch or hack nearly. And please do not take me wrongly, it's not about the solution itself, but rather meant as a comment to the package author. Additional transpilation of node_modules content is not a good long-term solution at all.

Depending on a package which does use lambda function is really not a great idea, especially when the functionality itself takes 10 lines of code.

This should be fixed by dropping the dependency on lazy-value and coding the same internally or depending on a package created by someone who actually knows how to create packages (e.g. produce minimized/transpiled outputs and not this nonsense).

As of now, the best solution is to downgrade to 0.9.8 and wait for this to be solved properly.

**Comment by warlord0** [comment]
@hambroz you're right it is a "work around". I haven't closed this issue for that reason.

The transpiling add a chunk of processing time to my build so running hmr it causes quite a delay. So whilst developing I don't use the plugin and only enable it when I need to test following a significant development.

If I were more confident I'd have a go at fixing it within `lazy-value` as I suspect that's where it isn't transpiling correctly due to the arrow function.

**Comment by hambroz** [comment]
@warlord0 Completely agree with your opinion on this!

As of `lazy-value`, wouldn't [https://babeljs.io/repl](https://babeljs.io/repl) and [https://jscompress.com/](https://jscompress.com/) do the trick? Of course, the best would be to setup `babel` for the project and do this when publishing the package, i.e. the same way how @xkjyeah does this for this package.

But as said before, this functionality is really rather trivial, so I would personally include it in the sources of `vue2-google-maps` itself.

**Comment by xkjyeah** [comment]
@Jan

Feel free to do a PR. Agreed that the functionality is simple. I considered
writing my own lazy-value, but saw this package and decided to use it.

The deeper philosphical question is what if there were many other
dependencies that all required transpilation? and to me it shouldn't be the
package maintainer's responsibility to duplicate code or write in 5yr old
JS syntax to support IE11.

IMO if you have a requirement to support an old browser, shouldn't it be
your responsibility to override the babel-loader config for individual
packages?

Example: I'm working on automated tests for this package, and to support an
old browser such as PhantomJS, I had to write this:

https://github.com/xkjyeah/vue-google-maps/blob/karma-testing/webpack.config.js#L42

And personally I prefer this solution, even if it means that many more
people's webpack.config.js have to be changed.

After all, to support IE11, you also need to do stuff like import
babel-polyfill.

LMK what you think.

**Comment by jan** [comment]
@xkjyeah Sorry, can't help you with this. ;-)

**Comment by warlord0** [comment]
@xkjyeah I hate IE11 with a passion... but sadly many organisation including our own still use Windows 7 for corporate applications.

I completely understand your point of view and actually that seems a more elegant work around that transpiling ALL of my JS, just because of lazy-value. Maybe it'll even cut down the hmr load time for me.

**Comment by xkjyeah** [comment]
You could manually require lazy-value to be transpiled (see the link to the
code in the karma-test branch)

**Comment by hambroz** [comment]
@xkjyeah Thanks, I will try to find some time and create the PR.

About your question, if it was a matter of code duplication, I would completely agree with your point of view. However, with tools like webpack/babel etc. everyone can easily do what you do with this package, e.g. provide files meant for distribution and the source files as separate entities.

TBH this is the first package I encountered so far that has problems with IE11. I am developing a Nuxt.js/Vue.js frontend application and transpiling my own code to support IE11, at the same time node_modules are never transpiled and everything works (of course, after including the polyfill you mentioned).

I would not mix polyfilling (e.g. solving missing browser functionality) and transpilation (e.g. missing JS language constructs). Yes, you need a polyfill for IE11 support, but you should not need to transpile published node modules, at least IMO. I might be completely mistaken though;)

**Comment by maritaria** [comment]
I'm working on a project that is using this library and needs to support IE, would love to have this release available asap 👍

**Comment by hambroz** [comment]
As agreed, should be solved in #461. If anything, let me know.

**Comment by hambroz** [comment]
@xkjyeah Thank you very much for publishing **v0.10.2**. Everything works as expected after updating!

**Comment by snoopdouglas** [comment]
This was *still* causing me problems with Vue CLI 3.

I have no idea what it is that is causing the files in `dist/` to be compiled into `chunk-vendors` with spreads (when the compiled files in `dist/` seemed to have them removed) - but this appears to fix the issue:

```js
// vue.config.js
module.exports = {
  transpileDependencies: [/\bvue2-google-maps\b/],
};
```

This clearly isn't an optimal solution, though; something else is going on here.

Pinging those who've found #452 as well.

**Comment by maritaria** [comment]
Would it be possible for you to figure out what dependency container spread operators in your build?
You can do this by doing something along the lines of:
- build your app without compression/modification (devbuild)
- run the app on ie-whatever
- see what line the error occurs at (probably an eval)
- look at the compiled webpack output and find the module id
- in a modern browser (ff/chrome) use source maps to expand ://webpack-internal/modules/{id} to find original module
